### PR TITLE
Remove non-stub code from pyis

### PR DESCRIPTION
**Pull request by alvarocaceres**
Also removed @abstractmethod decorators from codecs stub

### CODE REVIEW

File: stdlib/2.7/codecs.pyi [hunk 1]
```diff
@@ -115,9 +115,9 @@ class Codec:
         ...
 
 class IncrementalEncoder:
+    errors = ...  # type: str
     def __init__(self, errors: str = ...) -> None:
-        self.errors = errors
-    @abstractmethod
```
Comment (JukkaL) [inline stdlib/2.7/codecs.pyi]: `@abstractmethod` can actually be important, if the method really is supposed to be abstract. Why did you remove it?

**Comment by matthiaskramm** [comment]
So in other words, an `@abstractmethod` in a `pyi` says: This method is there, but you can't call it?

How does mypy use this information (compared to leaving that method out altogether)? Checking validity of subclasses? Or for MRO lookup?

**Comment by gvanrossum** [comment]
It means you can't instantiate the class. Nor can you instantiate a
subclass that doesn't have a concrete implementation of all abstract
methods.

File: stdlib/2.7/codecs.pyi (hunk 1 above)
Comment (alvarocaceres) [inline stdlib/2.7/codecs.pyi]: main reason was pytype compatibility -- the way I justified it to myself was that the implementation doesn't use the actual abc.abstractmethod, but I can see how that's debatable...

**Comment by matthiaskramm** [comment]
That *is* a useful concept to type-check. Alvaro, could you restore the `@abstractmethod` decorators?

**Comment by alvarocaceres** [comment]
added back -- sorry, shouldn't have gone together with the other changes in here in the first place
